### PR TITLE
NETOBSERV-1692: Add FLP-based filters & deduper options

### PR DESCRIPTION
**Pull request by jotak**
## Description

FLP-based dedup allows to decrease Loki CPU / memory / storage a lot (~50%) at the cost of minimal loss in data accuracy (e.g. loosing  interfaces involved in egress traffic)

Filters API allow to customize FLP filters

~~TODO:~~

- [x] Reverse filtering logic (currently filters == remove, should be filters == keep)
- [x] Add sampling config in filters
- [x] Manage numeric fields (in a limited way)
- [x] Manage regex (needs FLP change)

## Dependencies

- https://github.com/netobserv/flowlogs-pipeline/pull/740

## Checklist

If you are not familiar with our processes or don't know what to answer in the list below, let us know in a comment: the maintainers will take care of that.

* [x] Is this PR backed with a JIRA ticket? If so, make sure it is written as a title prefix _(in general, PRs affecting the NetObserv/Network Observability product should be backed with a JIRA ticket - especially if they bring user facing changes)._
* [x] Does this PR require product documentation?
  * [ ] If so, make sure the JIRA epic is labelled with "documentation" and provides a description relevant for doc writers, such as use cases or scenarios. Any required step to activate or configure the feature should be documented there, such as new CRD knobs.
* [ ] Does this PR require a product release notes entry?
  * [ ] If so, fill in "Release Note Text" in the JIRA.
* [ ] Is there anything else the QE team should know before testing? E.g: configuration changes, environment setup, etc.
  * [ ] If so, make sure it is described in the JIRA ticket.
* QE requirements (check 1 from the list):
  * [x] Standard QE validation, with pre-merge tests unless stated otherwise.
  * [ ] Regression tests only (e.g. refactoring with no user-facing change).
  * [ ] No QE (e.g. trivial change with high reviewer's confidence, or per agreement with the QE team).

### CODE REVIEW

File: apis/flowcollector/v1beta1/flowcollector_types.go [hunk 1]
```diff
@@ -505,13 +505,41 @@ type FlowCollectorFLP struct {
...
 	// +optional
 	Debug DebugConfig `json:"debug,omitempty"`
 }
 
+type FLPDeduperMode string
+
+const (
+	FLPDeduperDisabled FLPDeduperMode = "Disabled"
+	FLPDeduperDrop     FLPDeduperMode = "Drop"
+	FLPDeduperSample   FLPDeduperMode = "Sample"
```
Comment (jotak) [inline apis/flowcollector/v1beta1/flowcollector_types.go:524]: @jpinsonneau a possibility could be to add a "Merge" mode here that would involve infinispan like in your PoC

Comment (jpinsonneau) [reply apis/flowcollector/v1beta1/flowcollector_types.go:524]: Sounds good; but should we offer all of these or just support one or two modes in the end ?

Comment (jotak) [reply apis/flowcollector/v1beta1/flowcollector_types.go:524]: I see pros and cons on every mode and no clear "winner"

**Review by jpinsonneau — APPROVED** [review]
LGTM, thanks @jotak

**Pull request merged**